src/intlFormatDistance/index.ts

```diff
@@ -0,0 +1,282 @@
+ * | Distance between dates | Result               |
+ * |------------------------|----------------------|
+ * | 1 day                  | in 1 Tag             |
+ * | 1 week                 | in 1 Woche           |
+ *
+ *
+ *  With `options.numeric: 'auto' `
+ * | Distance between dates | Result               |
+ * |------------------------|----------------------|
+ * | 0 seconds              | now                  |
+ * | 1 day                  | tomorrow             |
+ *
+ *
+ *
+ * | Other Options     | Possible values         |  Result            |
+ * |-------------------|-------------------------|--------------------|
+ * | localeMatcher     | 'lookup' and 'best fit' | N/A                |
+ * |                   |                         |                    |
+ * | numeric           | 'always'                | 1 day ago          |
+ * |                   | 'auto'                  | yesterday          |
+ * |                   |                         |                    |
+ * | style             | 'long'                  | in 1 month         |
+ * |                   | 'short'                 | in 1 mo.           |
+ * |                   | 'narrow'                | in 1 mo.           |
+ *
+ *
+ *
+ * @param {Date|Number} date - the date
+ * @param {Date|Number} baseDate - the date to compare with.
+ * @param {Object} [options] - an object with options.
+ * @param {String} [options.unit] - formats the distance with the given unit ('year', 'quarter', 'month', 'week', 'day', 'hour', 'minute', 'second').
+ * @param {String|String[]} [options.locale] - the locale to use (BCP 47 language tag). [For the reference see MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation).
+ * @param {String} [options.localeMatcher='best fit'] - the locale matching algorithm to use. Other value: "lookup".
+ * @param {String} [options.numeric='always'] - the output message format. Other value: "auto".
+ * @param {String} [options.style='long'] - the length of the internationalized message. Other values: "short" or "narrow";
+ * @returns {String} the distance in words according to language-sensitive relative time formatting.
+ * @throws {TypeError} 2 arguments required
+ * @throws {RangeError} `date` must not be Invalid Date
+ * @throws {RangeError} `baseDate` must not be Invalid Date
+ * @throws {RangeError} `options.unit` must not be invalid Unit
+ * @throws {RangeError} `options.locale` must not be invalid locale
+ * @throws {RangeError} `options.localeMatcher=` must not be invalid localeMatcher
+ * @throws {RangeError} `options.numeric=` must not be invalid numeric
+ * @throws {RangeError} `options.style=` must not be invalid style
+ *
+ * @example
+ * // What is the distance between Apr, 4 1986 11:30:00 and Apr, 4 1986 10:30:00 in Intl?
+ * const result = intlFormatDistance(
+ *   new Date(1986, 3, 4, 11, 30, 0),
+ *   new Date(1986, 3, 4, 10, 30, 0));
+ * // => in 1 hour
```

Comment:
Please here and below 1) make output consistent (always has `=>`, no space between it and the comment) and 2) wrap strings into quotes.

```suggestion
 * //=> 'in 1 hour'
```